nestkernel/source_table.cpp

```diff
@@ -159,34 +159,35 @@ nest::SourceTable::clean( const thread tid )
           syn_id < ( *sources_[ tid ] ).size();
           ++syn_id )
     {
-      if ( ( *sources_[ tid ] )[ syn_id ] != NULL )
+      if ( ( *sources_[ tid ] )[ syn_id ] == NULL )
       {
-        if ( max_position.syn_id == syn_id )
+        continue;
+      }
+      std::vector< Source >*& sources = ( *sources_[ tid ] )[ syn_id ];
+      if ( max_position.syn_id == syn_id )
+      {
+        // we need to add 2 to max_position.lcid since
+        // max_position.lcid + 1 can contain a valid entry which we
+        // do not want to delete.
+        if ( max_position.lcid + 2 < static_cast< long >( sources->size() ) )
```

Comment:
would it make sense to introduce a const variable for `2` here and give it a meaningful name?